src/hooks/use-media.js

```
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+
+// from https://github.com/streamich/react-use/blob/master/src/useMedia.ts
+
+const isClient = typeof window === 'object';
+
+const useMedia = (query, defaultState = false) => {
+  const [state, setState] = useState(defaultState);
+
+  useEffect(() => {
```

The SSR problem we saw with the Navbar is something we had also seen when building out the DevHub. The issue stems from Gatsby not having a `window` object at build time, causing media queries/hooks to fail silently during the SSR process and not properly be re-hydrated on mount. This is also an [existing issue the react-responsive library has](https://github.com/contra/react-responsive/issues/162) and so using this library as-is isn't enough to get another check of the window during a production build.

The solution we had for DevHub (and also referenced in this issue) is embodied in this hook we took from DevHub:
- We use a `useEffect` to run on mount
- *We explicitly check for a `window`*
- We add a listener to the window for the media query so resizing works as expected.

The `react-responsive` `useMediaQuery` suggests hardcoding options for a specific screen size, but does not check on the mount, so we were stuck with one state. I could not get the intended behavior with the library so I decided to bring in this hook.

What do you think about this? Have you seen other issues when using the `useMediaQuery` hook?

---

To be honest, we don't use `useMediaQuery` widely throughout the site at the moment, so I don't have any experience with it. But now I'm inclined to do some investigating... 🕵️‍♀️ 

I have, however, run into this "window is not defined" issue in other ways and like this approach. I have no allegiance to `react-responsive` so am happy with this!

---

To be honest, we don't use useMediaQuery widely throughout the site at the moment, so I don't have any experience with it. But now I'm inclined to do some investigating... 🕵️‍♀️

I have, however, run into this "window is not defined" issue in other ways and like this approach. I have no allegiance to react-responsive so am happy with this!